marin/environments/swe_bench_env.py

```
@@ -0,0 +1,110 @@
+        split = self._train_split if mode == "train" else self._test_split
+        example = random.choice(split)
+
+        instance_id: str = example["instance_id"]
+        prompt: str = example["text"]
+
+        # Run inference with the model to generate the code patches
+        inference_payload = self.client.chat.completions.create(
+            model=self.model_name,
+            messages=[{"role": "user", "content": prompt}],
+        )
+        completion = inference_payload.choices[0].message.content
+        patch: str = extract_diff(completion)
+
+        # Write out the predictions/patches for SWE-Bench to evaluate
+        record = {
+            "model_name_or_path": self.model_name,
+            "instance_id": instance_id,
+            "full_output": completion,
+            "model_patch": patch,
+        }
+        predictions_path: str = os.path.join(output_path, "predictions.jsonl")
+        with open(predictions_path, "w") as f:
+            f.write(json.dumps(record) + "\n")
+
+        # Evaluate the patches generated by the model
+        commands = [
+            "python",
+            "-m",
+            "swebench.harness.run_evaluation",
+            "--run_id",
+            step_id,
+            "--dataset_name",
+            self.dataset_name,
+            "--split",
+            mode,
+            "--predictions_path",
+            predictions_path,
+            "--instance_ids",
+            instance_id,
+            "--max_workers",
+            str(self.max_workers),
+            "--report_dir",
+            output_path,
+        ]
+        process = subprocess.Popen(commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        # Award a reward of 1 when GitHub issues is resolved
+        reward: float = 0
+        if "Instances resolved: 1" in stdout.decode("utf-8", errors="ignore"):
+            reward = 1
```

this could easily lead to silent failures / make it hard to debug - can we check the format of the output?  We probably should store the stdout of the run just for debugging